app/src/main/java/me/quadphase/qpdex/pokedex/PokedexManager.java

```diff
@@ -0,0 +1,163 @@
+package me.quadphase.qpdex.pokedex;
+
+import android.app.Application;
+import android.content.Context;
+import android.content.res.AssetFileDescriptor;
+import android.widget.ArrayAdapter;
+
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+import me.quadphase.qpdex.IntroActivity;
+import me.quadphase.qpdex.pokedex.CentralAudioPlayer;
+import me.quadphase.qpdex.pokemon.MinimalPokemon;
+import me.quadphase.qpdex.pokemon.Pokemon;
+import me.quadphase.qpdex.pokemon.Type;
+
+/**
+ * High level manager of all Pokemon related information
+ * This class is a singleton that should be use when interaction is needed with activities.
+ */
+//TODO: Document with JavaDocs
```

Comment:
I would seem I cannot successfully code while Documenting xD
That's why I wiped out this class and started from scratch yesterday.